wallet/src/repository/wallet/tests/factories.rs

```diff
@@ -0,0 +1,129 @@
+use std::{cell::RefCell, rc::Rc};
+
+use super::*;
+
+pub fn wallet(data: Option<HashMap<Vec<u8>, Vec<u8>>>) -> (Wallet<db::HashMapDb>, db::HashMapDb) {
+    let params = params::Params::default();
+    let mnemonic = types::MnemonicGen::new()
+        .with_len(types::MnemonicLength::Words12)
+        .generate();
+    let source = types::SeedSource::Mnemonics(mnemonic);
+    let master_key = crypto::gen_master_key(
+        params.seed_password.as_ref(),
+        params.master_key_salt.as_ref(),
+        &source,
+    )
+    .unwrap();
+    let engine = types::SignEngine::signing_only();
+    let default_account_index = 0;
+    let default_account =
+        account::gen_account(&engine, default_account_index, &master_key).unwrap();
+
+    let mut rng = rand::rngs::OsRng;
+    let salt = crypto::salt(&mut rng, params.db_salt_length);
+    let iv = crypto::salt(&mut rng, params.db_iv_length);
+
+    let storage = Rc::new(RefCell::new(data.unwrap_or_default()));
+    let db = db::HashMapDb::new(storage);
+    let wallets = Wallets::new(db.clone());
+
+    // Create the initial data required by the wallet
+    wallets
+        .create(
+            &db,
+            types::CreateWalletData {
+                iv,
+                salt,
+                id: "test-wallet-id",
+                name: None,
+                caption: None,
+                account: &default_account,
+            },
+        )
+        .unwrap();
+
+    let wallet = Wallet::unlock(db.clone(), params, engine).unwrap();
+
+    (wallet, db)
+}
+
+pub fn pkh() -> types::PublicKeyHash {
+    let bytes: [u8; 20] = rand::random();
```

Comment:
Non-deterministic tests? :eyes:

Reply:
hahaha not really, these are pretty common for generating random data for tests, think of
[faker](https://github.com/paraseba/faker) or its Rust equivalent [faker-rs](https://github.com/cksac/fake-rs)